### DISCUSSION TITLE
Try Out New: No Code GUI Builder⚡️

### DISCUSSION THREAD

**Discussion by rigvedmaanas** [TomSchimansky/CustomTkinter#2417] [Showcase]
Hi All,

I'm excited to share my new project, [Custom Tkinter Builder](https://github.com/rigvedmaanas/CustomTkinterBuilder)!

This is a tool for anyone who wants to build GUIs for Python projects without needing to write tons of code. Instead of writing everything out, you can just drag and drop widgets and customize them with this software and generate the necessary code.
You can change the properties of all the widgets according to your preferences 🤩. This software has builtin `Icon` support which contains more than 2000 icons with varied color, size and type from [Material Design Icons](https://fonts.google.com/icons?icon.set=Material+Icons). This software will help you to create responsive and better looking UI fast ⚡. It’s free and cross-platform too.

# This is the how the software looks.

![Screenshot 2024-05-17 at 8 13 37 PM](https://github.com/TomSchimansky/CustomTkinter/assets/77579661/0d003013-289b-4a3c-b931-9228990aca22)

# Examples created using Custom Tkinter Builder

### Dark Mode
![Dark Mode - Login](https://github.com/rigvedmaanas/CustomTkinterBuilder/assets/77579661/133ef73f-3b55-4181-bf78-a4ba1e9b19a3)
### Light Mode
![Light Mode - Login](https://github.com/rigvedmaanas/CustomTkinterBuilder/assets/77579661/482ed8c3-0573-4103-9aa6-45d0258eacdf)

### Dark Mode
![Dark Mode - Spotify](https://github.com/rigvedmaanas/CustomTkinterBuilder/assets/77579661/28970507-3125-4cd8-bb42-d6a7d3fee359)
### Light Mode
![Light Mode - Spotify](https://github.com/rigvedmaanas/CustomTkinterBuilder/assets/77579661/eb2a3b08-02ac-4bea-bf5f-9da723bb6101)

## Video Library
![Youtube](https://github.com/rigvedmaanas/CustomTkinterBuilder/assets/77579661/612572d7-456c-45ef-aeff-ea1108cbf995)

# Available Widgets

- Frame
- Button
- Label
- Entry
- Vertical Scrollable Frame
- Horizontal Scrollable Frame
- Switch
- Check Box
- Radio Button
- Segmented Button
- Horizontal Slider
- Vertical Slider
- Combo Box
- Option Menu
- Text Box
- Progress Bar
- Vertical Scrollbar
- Horizontal Scrollbar

# Features

* ✅ **Drag and Drop** user interface
* ✅ Adjust **all the parameters** of all the widgets.
* ✅ Supports responsive **Pack** method
* ✅ Supports **15/16 core widgets**
* ✅ Includes **10 pre customized** widgets
* ✅ Includes **inbuilt icon** support (more than **2000** ready to use icons with 5 different styles and customisable color)
* ✅ Includes **Palette** which allows the users to change the colors of a UI with a single click of a button.
* ✅ Supports **Scrollable Frame** (Vertical and Horizontal)
* ✅ Export to .py (OOP and standard)
* ✅ Save project (.json)
* ✅ Duplicate widgets, Delete Widgets, Change order of widgets and Change parent of widgets
* ✅ Visualize the structure of your UI with the inbuilt **Hierarchy Panel**
* ✅ Allows users to work on both **light** and **dark** mode of the UI
* ✅ Add images, change fonts, etc

# Download
<br> <p align='center'> [<img src="https://img.shields.io/badge/Download-Custom%20Tkinter%20Builder-informational?&logo=python&logoColor=yellow&color=green" width="350">](https://github.com/rigvedmaanas/CustomTkinterBuilder/releases/tag/v1.0.0)  </br>

The software is available for both Windows (.exe) and Mac (.dmg).
Official Documentation is available at [https://github.com/rigvedmaanas/CustomTkinterBuilder/wiki](https://github.com/rigvedmaanas/CustomTkinterBuilder/wiki)

If you're looking for a simpler way to build Python GUIs, check out [Custom Tkinter Builder](https://github.com/rigvedmaanas/CustomTkinterBuilder)! I'd love to hear your thoughts and feedback.

**Comment by LousyBook-01** [comment]
cool

**Comment by purplemanta6754** [comment]
tysm @rigvedmaanas 💖

**Comment by rigvedmaanas** [comment]
Thank you guys!!!. We got 100 stars 🥳🥳🥳
**Edit: Over 240 stars 🚀**

**Comment by LiKTOi1029** [comment]
This is exactly what I needed right now thanks a bunch!
